coderd/workspaces.go

```diff
@@ -1029,6 +1029,55 @@ func (api *API) watchWorkspace(rw http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// @Summary Transfer workspace owner by ID
+// @ID transfer-workspace-owner-by-id
+// @Security CoderSessionToken
+// @Accept json
+// @Tags Workspaces
+// @Param workspace path string true "Workspace ID" format(uuid)
+// @Param request body codersdk.TransferWorkspaceOwnerRequest
+// @Success 204
+// @Router /workspaces/{workspace}/transfer [put]
+func (api *API) putTransferWorkspace(rw http.ResponseWriter, r *http.Request) {
+	var (
+		ctx               = r.Context()
+		workspace         = httpmw.WorkspaceParam(r)
+		auditor           = api.Auditor.Load()
+		aReq, commitAudit = audit.InitRequest[database.Workspace](rw, &audit.RequestParams{
+			Audit:   *auditor,
+			Log:     api.Logger,
+			Request: r,
+			Action:  database.AuditActionWrite,
+		})
+	)
+
+	defer commitAudit()
+	aReq.Old = workspace
+
+	var req codersdk.TransferWorkspaceOwnerRequest
+	if !httpapi.Read(ctx, rw, r, &req) {
+		return
+	}
+
+	now := database.Now()
+	if err := api.Database.UpdateWorkspaceOwnerByID(ctx, database.UpdateWorkspaceOwnerByIDParams{
+		ID:        workspace.ID,
+		OwnerID:   req.OwnerID,
```

Comment:
Do we need any server-side validation for values in `TransferWorkspaceOwnerRequest`?